File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> vals;
        vector<int> out;

        for(int i = 0; i < nums.size(); i++){

            int curr_val = nums[i];
            int diff = target - curr_val;

            if(!vals.count(curr_val)){
                vals[curr_val] = i;
            }
            if(vals.count(diff) && (vals[diff] != i)){
                out.push_back(vals[diff]);
                out.push_back(i);
                return out; 
            }

        }

        return out; 
    }
};

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-3.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        total = sum(nums)

        if total % 2:
            return False
        
        target = total // 2

        dp = [False] * (target + 1)
        dp[0] = True

        for n in nums:
            for j in range(len(dp)-1, n-1, -1):
                dp[j] = dp[j] | dp[j - n]
        
        return dp[target]